chi_analysis/add_systematics_qbh.py

```diff
@@ -19,48 +19,188 @@
+##     fastNloPlot2=fastNloPlot2.Rebin(len(chi_binnings[1])-1,"fastNloQCD#chi4800_13000_rebin1",chi_binnings[1])
+##     fastNloFinalPlots=[fastNloPlot1,fastNloPlot2]
+##     fastNloFinalPlotsBackup=[]
 
-crossSectionMap=[[4000,28.65327],[4500,10.28881],[5000,3.728508],[5500,1.343979],[6000,0.4758148],[6500,0.1634881],[7000,0.05385596],[7500,0.01678352],[8000,0.004871688],[8500,0.001292072],[9000,0.0003054339],[9500,0.00006221544],[10000,0.00001040396],[10500,0.000001327101],[11000,0.0000001145733]]
-totNEvents={}
-totNEvents[6500]=999977
-totNEvents[7000]=999973
-totNEvents[7500]=999970
-totNEvents[8000]=999964
-totNEvents[8500]=999962
-totNEvents[9000]=999957
-totNEvents[9500]=999936
+    fastNloPlots=[]
+    fastNloFile=TFile.Open("/afs/cern.ch/work/z/zhangj/private/CMSSW_7_4_15/src/fnl5662i_v23_fix_CT14_ak4.root")
+    ewkFile=TFile.Open("/afs/cern.ch/work/z/zhangj/private/CMSSW_7_4_15/src/DijetAngularCMS13_ewk.root")
+    fastNloMassBins=[[3600,4200],[4200,4800],[4800,5400],[5400,6000]]
+    for fastNloMass in fastNloMassBins:
+        fastnlo=fastNloFile.Get("chi-"+str(fastNloMass[0])+"-"+str(fastNloMass[1]))
+        ewk=ewkFile.Get("chi-"+str(fastNloMass[0])+"-"+str(fastNloMass[1]))
+        for b in range(fastnlo.GetXaxis().GetNbins()):
+            fastnlo.SetBinContent(b+1,fastnlo.GetBinContent(b+1)*fastnlo.GetBinWidth(b+1))
+            low_bin=ewk.FindBin(fastnlo.GetXaxis().GetBinLowEdge(b+1))
+	    up_bin=ewk.FindBin(fastnlo.GetXaxis().GetBinUpEdge(b+1))
+	    correction=ewk.Integral(low_bin,up_bin-1)/(up_bin-low_bin)
+            fastnlo.SetBinContent(b+1,fastnlo.GetBinContent(b+1)*correction)
+        fastNloPlots.append(fastnlo)
+    fastNloFinalPlots=[]
+    h1=fastNloPlots[0]
+    h1.Scale(600.)
+    h2=fastNloPlots[1]
+    h2.Scale(600.)
+    h3=fastNloPlots[2]
+    h3.Scale(600.)
+    h4=fastNloPlots[3]
+    h4.Scale(600.)
+    fastNloPlot1=h1.Clone("fastNloQCD#chi3600_4200_rebin1")
+    fastNloPlot2=h2.Clone("fastNloQCD#chi4200_4800_rebin1")
+    h3.Add(h4)
+    fastNloPlot3=h3.Clone("fastNloQCD#chi4800_13000")
+    fastNloPlot3=fastNloPlot3.Rebin(len(chi_binnings[2])-1,"fastNloQCD#chi4800_13000_rebin1",chi_binnings[2])
+    fastNloFinalPlots=[fastNloPlot1,fastNloPlot2,fastNloPlot3]
+    fastNloFinalPlotsBackup=[]   
 
-massbins=[[4200,4800],[4800,13000]]
-slopes=[0.04,0.05]
+if isRS1QBH:
+    crossSectionMap=[[3500,0.4288],[4000,0.1464],[4500,0.05148],[5000,0.01829],[5500,0.006472],[6000,0.002250],[6500,0.0007599],[7000,0.0002461]]
+    outname="datacard_shapelimit13TeV_QBH_"+energy+"_RS1_chi_v1.root"
+    qbhfile=TFile.Open("root://eoscms//eos/cms/store/group/dpg_trigger/comm_trigger/L1Trigger/Jingyu/samples/DiJet/pythia_qbh_"+energy+"_RS1_v1.root")
+    print "root://eoscms//eos/cms/store/group/dpg_trigger/comm_trigger/L1Trigger/Jingyu/samples/DiJet/pythia_qbh_"+energy+"_RS1_v1.root"
+    signalHistFileName="qbh_"+str(energy)+"_RS1_chi_v1.root"
+    massbins=[[3600,4200],[4200,4800],[4800,13000]]
```

Comment:
This is duplication of the code above. Rather only put stuff that is different in the if statement and everything else outside.